### PR TITLE
Adding Storefront Catalog to base parameters for SFCC app.

### PR DESCRIPTION
**Pull request by jgrahamdev**
## Purpose

When picking products/categories from SFCC, a Storefront Catalog serves as the root for products/categories that can be used on the site.

## Approach

Updated apps `onConfigure` hook to validate Salesforce config parameters, then we fetch the storefront catalog and add the corresponding id to app's installation parameters. We then use this id when searching for products or categories to make sure we're only retrieving relevant results for the given site.

## Breaking Changes

If products/categories were selected previously that were not a child of the Storefront catalog, those links will no longer be resolved. Additionally, the field value for categories has been changed from a combination of catalog and category id to just the category id, since the catalog is now consistent.

### CODE REVIEW

File: apps/salesforce-commerce-cloud/src/components/field/ItemCard.tsx [hunk 1]
```diff
@@ -32,13 +32,13 @@ const ItemCard = (props: ItemCardProps) => {
   const installParameters = sdk.parameters.installation as AppInstallationParameters;
   const client = new SfccClient(installParameters);
 
-  const [itemId, catalogId] = props.id.split(':');
+  // const [itemId, catalogId] = props.id.split(":")
```
Comment (lilbitner) [inline apps/salesforce-commerce-cloud/src/components/field/ItemCard.tsx:35]: Perhaps can remove this commented out code?

File: apps/salesforce-commerce-cloud/src/components/field/SelectItemAction.tsx [hunk 2]
```diff
@@ -39,13 +39,11 @@ const SelectItemAction = (props: { fieldValue?: string | string[] }) => {
   const client = new SfccClient(installParameters);
   const currentItemQueries = useQueries({
     queries: queryArray.map((id: string) => {
-      const [itemId, catalogId] = id.split(':');
+      // const [itemId, catalogId] = id.split(":")
```
Comment (lilbitner) [inline apps/salesforce-commerce-cloud/src/components/field/SelectItemAction.tsx:42]: perhaps can remove this commented out code as well?

File: apps/salesforce-commerce-cloud/src/locations/ConfigScreen.tsx [hunk 3]
```diff
@@ -3,17 +3,22 @@ import { ConfigAppSDK } from '@contentful/app-sdk';
...
-  siteId: string;
+import { SalesforceConfig, SfccClient } from '../utils/Sfcc';
+
+// interface SalesforceConfig {
+//   clientId: string;
+//   clientSecret: string;
+//   organizationId: string;
+//   shortCode: string;
+//   siteId: string;
+// }
```
Comment (lilbitner) [inline apps/salesforce-commerce-cloud/src/locations/ConfigScreen.tsx:14]: same comment here, and throughout this file, around commented out code :)

File: apps/salesforce-commerce-cloud/src/locations/ConfigScreen.tsx [hunk 4]
```diff
@@ -156,7 +171,7 @@ const ConfigScreen = () => {
           <FormInput
             key={input.id}
             input={input}
-            value={parameters[input.id as keyof AppInstallationParameters]}
+            value={parameters[input.id as keyof AppInstallationParameters]!}
```
Comment (lilbitner) [inline apps/salesforce-commerce-cloud/src/locations/ConfigScreen.tsx:174]: is there perhaps a better fallback option here as opposed to enforcing it using `!` ? There may not be, but curious around the context here

File: apps/salesforce-commerce-cloud/src/utils/Sfcc.tsx [hunk 5]
```diff
@@ -1,5 +1,14 @@
 import axios, { AxiosInstance, InternalAxiosRequestConfig } from 'axios';
-import { AppInstallationParameters } from '../locations/ConfigScreen';
+// import { AppInstallationParameters } from "../locations/ConfigScreen"
```
Comment (lilbitner) [inline apps/salesforce-commerce-cloud/src/utils/Sfcc.tsx:2]: few cleanup lines here and line 13 as well!

**Review by lilbitner — COMMENTED** [review]
Hello @jgrahamdev! Thank you for contributing these changes and I apologize for the delayed review on behalf of the Integrations team here at Contentful. I just left very minor comments and clarifications here. Additionally, when running this code locally, since there are breaking changes, it looks like the only way to solve those breaking changes is to uninstall and reinstall the app. Is that correct? 

Additionally since this PR has been open for some time, certainly worth rebasing. And the build errors you see here are most likely not an issue of your own, but an issue of authorization for external contributors. This unfortunately is a manual process we are working on, but I am happy to run the build to see if it passes, once the PR is rebased and comments are addressed.

**Comment by whitelisab** [comment]
Confirmed with @jgrahamdev that we will close this PR for now and we can revisit this enhancement for the SFCC app in the future.

**Pull request closed without merging**